### PR TITLE
refactor: Update JSDoc signature(s) + make some internal methods static

### LINKED ISSUE #14: JSDoc comments don't reflect current function signatures
https://github.com/discordjs/voice/blob/a124f95f2eff187f6aba0cc2c506fac84c1444bd/src/DataStore.ts#L35-L37
The parameters in the comment differ from the function definition.

Consider adding [`eslint-plugin-tsdoc`](https://github.com/microsoft/tsdoc/tree/master/eslint-plugin#eslint-plugin-tsdoc) to a linting workflow to automate discovery of such mistakes.

### PR DESCRIPTION
**Pull request by alii**
Fixes #14 on the Project board

**Please describe the changes this PR makes and why it should be merged:**
This PR...
• Adds `eslint-plugin-jsdoc` as well
• Fixes the current JSDoc signatures.
• Makes a couple of internal methods static
• Adds a typecheck step to the GitHub test workflow action

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added) (internal only)

### CODE REVIEW

File: src/DataStore.ts [hunk 1]
```diff
@@ -31,7 +32,7 @@ export interface JoinConfig {
 /**
  * Sends a voice state update to the main websocket shard of a guild, to indicate joining/leaving/moving across
  * voice channels
- * @param voiceChannel The voice channel to move to
+ * @param config
```
Comment (kyranet) [inline src/DataStore.ts:35]: Missing parameter description, I was thinking of this one:
```suggestion
 * @param config The voice state data to send to Discord
```

File: src/VoiceConnection.ts [hunk 2]
```diff
@@ -84,6 +83,7 @@ export class VoiceConnection extends EventEmitter {
 	 * Creates a new voice connection.
 	 *
 	 * @param joinConfig The data required to establish the voice connection
+	 * @param options Options to create a connection
```
Comment (kyranet) [inline src/VoiceConnection.ts:86]:
```suggestion
	 * @param options The connection options
```

File: src/VoiceConnection.ts [hunk 3]
```diff
@@ -401,6 +401,7 @@ export class VoiceConnection extends EventEmitter {
 /**
  * Creates a new voice connection
  * @param joinConfig The data required to establish the voice connection
+ * @param options Options to create a connection
```
Comment (kyranet) [inline src/VoiceConnection.ts:404]: The creation of a connection is already implied by the function's description on line 402.
```suggestion
 * @param options The connection options
```

File: src/networking/VoiceWebSocket.ts [hunk 4]
```diff
@@ -32,6 +32,7 @@ export class VoiceWebSocket extends WebSocket {
 	/**
 	 * Creates a new VoiceWebSocket
 	 * @param address The address to connect to
+	 * @param debug Whether or not we should print debug logs
```
Comment (kyranet) [inline src/networking/VoiceWebSocket.ts:35]: This does not print anything, it just enables the emission of debug events.
```suggestion
	 * @param debug Whether or not debug logs should be emitted
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -41,10 +41,14 @@
     "discord-api-types": "^0.12.1",
     "eslint": "^7.15.0",
     "eslint-config-marine": "^7.2.0",
+    "eslint-plugin-tsdoc": "^0.2.11",
     "typescript": "^4.2.0-dev.20210204"
   },
   "eslintConfig": {
     "extends": "marine",
+    "plugins": [
+      "eslint-plugin-tsdoc"
```
Comment (amishshah) [inline package.json:50]: I'm not entirely sure this is set up correctly? If I try to revert line 34/35 in `src/DataStore.ts` back to `@param voiceChannel ...` then no linter errors appear.

**Review by amishshah — CHANGES_REQUESTED** [review]

**Comment by iCrawl** [comment]
For the next time check if the issue is already assigned to a team member: 
image

**Comment by alii** [comment]
This is in an unmergable state right now, as `eslint-plugin-tsdoc` does not recognise the `@event` parameter. I've removed them for the time being, however I'm not sure what the correct replacement would be with the comments.

It's possible to properly type the EventEmitter with an interface or something like [typed-emitter](https://github.com/andywer/typed-emitter). Should I implement that?

**Comment by iCrawl** [comment]
https://tsdoc.org/pages/packages/tsdoc-config/

**Comment by alii** [comment]
So looks like the plugin recognizes it just fine, however, it's not playing nicely with the type specified:

Will look into it further

image

**Pull request closed without merging**